frontend/src/components/UserSearch.vue

```diff
@@ -138,8 +141,31 @@
 
 <script>
 import UsersRepository from "@/api/UsersRepository";
+import ClientsRepository from "@/api/ClientsRepository";
 import organizations from "@/assets/organizations"
 
+// REVIEW - Is this the best approach to define custom objects using Vue?
+//   Should the custom object be defined in a more Vue-compliant way?
```

Comment:
I think you're the first to define a class in the project that isn't just a helper class for Testcafe.

I don't have any experience with this, so I'm not sure what to say. A question though, was this done just so that we could have the `toString` method, which is just to show the items in the drop-down list?

Reply:
Yes, it needs an override of `toString()` but it is defined to store the Client ID and Role Name required to query the users against the UMS API. I have been reading the Vue documentation but it talks mostly about Components, which might not be the case here. Anyways, it seems like the entire Searcy By Role might need to be reworked a bit due to the current needs to support more advanced queries.

Reply:
> Anyways, it seems like the entire Searcy By Role might need to be reworked a bit due to the current needs to support more advanced queries.

You're right, but I think this Pull Request is still heading in the right direction. After that meeting we had, it seems we'll need to add:

* Ability to select multiple roles
* Ability to choose which columns to display

But we needn't add that in this PR. We can finish-up this PR's remaining issues (if we can), merge it, and carry on later. If the code was outright broken I'd be reluctant to merge it into master, but as is it's just missing needed features that we may not have tried to do all at once anyhow.